### PR TITLE
Define setup_transport method

### PR DESCRIPTION
**Pull request by Hannarst**
Using helper class and deferToThread and placeholder layer EchoLayer. Mostly copied (and simplified) from https://github.com/tgalal/yowsup/wiki/Yowsup-2.0-Sample-app

### CODE REVIEW

**Comment by Hannarst** [comment]
This is pretty much where I am currently, besides the teardown method. 

Known issues:
* It doesn't work sometimes. 

Everything seems to be getting set up correctly (no error messages) and getting to the yowstack main loop alright, but messages aren't received and obviously aren't echoed. It generally works again after a couple of restarts, and it works most of the time. 

Should I ignore this for now and carry on with actually connecting it to vumi stuff?

@smn @hodgestar

**Comment by Hannarst** [comment]
Also, some of the imports might be unnecessary now, but I'll clean that up a bit later.

File: vxyowsup/whatsapp.py [hunk 1]
```diff
@@ -1,10 +1,26 @@
...
+from yowsup.layers import YowLayerEvent
+from yowsup.stacks import YowStack, YowStackBuilder
+from yowsup import env
+
+import sys, argparse, yowsup, logging
+
+from yowsup.layers.interface                           import YowInterfaceLayer, ProtocolEntityCallback
+from yowsup.layers.protocol_messages.protocolentities  import TextMessageProtocolEntity
+from yowsup.layers.protocol_receipts.protocolentities  import OutgoingReceiptProtocolEntity
+from yowsup.layers.protocol_acks.protocolentities      import OutgoingAckProtocolEntity
```
Comment (hodgestar) [inline vxyowsup/whatsapp.py]: We tend to only leave on space before `import`, although this does make things quite readable.

File: vxyowsup/whatsapp.py [hunk 2]
```diff
@@ -39,3 +58,51 @@ def handle_outbound_message(self, message):
             return self.publish_nack(message['message_id'], 'failed')
         return self.publish_ack(message['message_id'], 
             'remote-message-id')
+            
+            
+    def print_error(self, f):
+        print f
+        return f
+            
+class Client:
```
Comment (hodgestar) [inline vxyowsup/whatsapp.py]: This should probably inherit from `object` (for backwards compatibility with ancient versions of Python, classes that don't inherit from anything in Python 2 are rather odd).

**Comment by hodgestar** [comment]
@Hannarst :+1: on your commits. If you're happy with my commits, you are welcome to merge.

**Comment by Hannarst** [comment]
:+1:  on your changes, @hodgestar

**Pull request merged**